release-24.2: roachtest: enable runtime assertions in nightlies without metamorphic constants

DarrylWong opened the pull request:
Backport:
  * 2/2 commits from "roachtest: enable runtime assertions in nightlies without metamorphic constants" (#128699)
  * 1/1 commits from "roachtest: reuse Github issues for runtime-assertions failures" (#131255)
  * 1/1 commits from "roachtest: reword github runtime assertion failure note" (#131368)

Please see individual PRs for details.

/cc @cockroachdb/release

Release justification: test only change

DarrylWong committed:
roachtest: enable assertions in nightlies without metamorphic constants

In #114618 we disabled runtime assertions and metamorphic constants due
to their flakiness.

Here we reenable them but with metamorphic constants disabled, as they
are the more disruptive of the two. We also change the framework to take
in a --metamorphic-cockroach-ea-probability instead of hardcoding it.

DarrylWong committed:
roachtest: route runtime assertion build timeouts to test-eng

As we determine the stability of runtime assertions, temporarily
route all timeout failures to test-eng to reduce noise for other
teams.

DarrylWong committed:
roachtest: reuse Github issues for runtime-assertions failures

Generally when a test has open failures for both builds with
and without runtime assertions, the failures are likely due to
the same (non runtime assertion related) issue.

After running tests with runtime-assertions metamorphically on
for a few weeks, we have yet to see this not be the case. To
reduce toil of tracking and closing extra issues, github issues
will now be reused between runtime assertion and regular builds.

Additionally, the release blocker label is no longer omitted for
runtime-assertions. This is due to the reuse and also the
stability of the builds in our test runs.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/131316)
<!-- Reviewable:end -->

DarrylWong commented:
[0.2 smoke test](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17022500?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true)

DarrylWong committed:
roachtest: reword github runtime assertion failure note

The old message mentioned that there may be a seperate issue
without the runtime-assertions label. This is no longer true
now that we reuse the issue and may be confusing. This change
rewords the message.

Fixes: none
Epic: none
Release note: none
